File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    answer = 0
    N = len(word)
    arr = []
    dictionary = [ 'A', 'E', 'I', 'O', 'U']
    def dfs(lst):
        if(len(dictionary)==len(lst)):
            return
        for i in range(len(dictionary)):
            w = lst + [dictionary[i]]
            arr.append("".join(w))
            dfs(w)
    dfs([])
    ## print(arr)
    print(arr.index(word))
    return arr.index(word)+1

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    var answer = 0;
    const arr = ['A', 'E', 'I', 'O', 'U']
    const N = arr.length
    let isFind = false
    function dfs(idx, lst){
        // console.log(lst.join(""),answer)
        if(isFind){return}
        answer++
        if(lst.join("") === word){
            console.log("####",lst, --answer)
            isFind=true
            return
        }
        if(lst.length == N){
            return
        }
        for(let idx=0;idx<N;idx++){
            dfs(idx+1, [...lst, arr[idx]])
        }
    }
    dfs(0,[])
    return answer;
}